### ISSUE TITLE
Feature Request: AWS::Lambda::Function - If `ZipFile` is used for the Lambda Code, the `Handler` needs to start with `index.*`

### ISSUE THREAD

**Issue by r-heimann** [aws-cloudformation/cfn-lint#4567]
### Is this feature request related to a new rule or cfn-lint capabilities?

New capability

### Describe the feature you'd like to request

Not 100% sure this is correct, but i believe that a rule for this would be very useful (just came across this while changing a Lambda from `S3Bucket/S3Key` to `ZipFile` and breaking it in the process).

### Describe the solution you'd like

When creating an `ZipFile` Lambda with CloudFormation

```yaml
  LambdaFunction:
    Type: AWS::Lambda::Function
    Properties:
      Role: !GetAtt LambdaRole.Arn
      Handler: index.lambda_handler
      Runtime: python3.14
      Architectures:
        - arm64 # CPU
      MemorySize: 128 # RAM
      Code:
        ZipFile: |

          def lambda_handler(event, context):
              print("Hello World")
```

the file that will be created inside the Lambda will always be (for python) `index.py` and the `Handler` needs to start with `index.*` or the Function will not work when executing it:

>[WARNING] 20...Z LAMBDA_WARNING: Unhandled exception. The most likely cause is an issue in the function code. However, in rare cases, a Lambda runtime update can cause unexpected function behavior. For functions using managed runtimes, runtime updates can be triggered by a function change, or can be applied automatically. To determine if the runtime has been updated, check the runtime version in the INIT_START log entry. If this error correlates with a change in the runtime version, you may be able to mitigate this error by temporarily rolling back to the previous runtime version. For more information, see https://docs.aws.amazon.com/lambda/latest/dg/runtimes-update.html
>[ERROR] Runtime.ImportModuleError: Unable to import module 'lambda': No module named 'lambda' Traceback (most recent call last):

### Additional context

When users change a Lambda source from `S3Bucket/S3Key` to `ZipFile`, they could make the mistake of not updating the `Handler`, which breaks the Lambda since it cannot find the file to execute.

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request

### Would this feature include a breaking change?

- [ ] ⚠️ This feature might incur a breaking change
